src/vscode-bicep/src/commands/decompileParams.ts

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import {
+  IActionContext,
+  IAzureQuickPickItem,
+} from "@microsoft/vscode-azext-utils";
+import vscode, { MessageItem, Uri, window } from "vscode";
+import { DocumentUri, LanguageClient } from "vscode-languageclient/node";
+import { OutputChannelManager } from "../utils/OutputChannelManager";
+import { Command } from "./types";
+import * as fse from "fs-extra";
+import * as path from "path";
+import { UserCancelledError } from "@microsoft/vscode-azext-utils";
+import assert from "assert";
+
+interface DecompileParamsCommandParams {
+  jsonUri: DocumentUri;
+  bicepPath: string | undefined;
+}
+
+interface DecompiledBicepparamFile {
+  contents: string;
+  absolutePath: string;
+}
+
+interface DecompileParamsCommandResult {
+  decompiledBicepparamFile: DecompiledBicepparamFile | undefined;
+  errorMessage: string | undefined;
+}
+
+export class DecompileParamsCommand implements Command {
+  public readonly id = "bicep.decompileParams";
+  public constructor(
+    private readonly client: LanguageClient,
+    private readonly outputChannelManager: OutputChannelManager
+  ) {}
+
+  public async execute(
+    context: IActionContext,
+    documentUri?: vscode.Uri | undefined
+  ): Promise<void> {
+    documentUri = documentUri ?? window.activeTextEditor?.document.uri;
+    if (!documentUri) {
+      throw new Error(
+        "Please open a JSON Parameter file before running this command"
+      );
+    }
+
+    const canDecompile =
+      await DecompileParamsCommand.mightBeArmParametersNoThrow(documentUri);
+    if (!canDecompile) {
+      this.outputChannelManager.appendToOutputChannel(
+        `Cannot decompile "${documentUri.fsPath}" into Bicep because it does not appear to be an ARM template parameter file.`
+      );
+      throw new UserCancelledError(
+        "Cannot decompile input because file provided is not a parameter file"
+      );
```

Comment:
It's a bit odd to be writing an error message to a channel and throwing a separate message, but it looks like we do the same in `decompile.ts`. What does the UI look like for this code path?

Reply:
`UserCancellationError` doesn't show anything on the UI (it is just used here to stop the execution of the command). We could also just throw a regular error and remove duplication.

Reply:
No need; this approach makes sense. My only concern would be if we were showing two error messages within the UI